### PR TITLE
feat(storage transfer): Added samples for storage transfer

### PR DESCRIPTION
**Pull request by thiyaguk09**
- [ ] Add quickstart samples for sts
- [ ] Add nearline samples for sts
- [ ] Add posix and manifest samples for sts
- [ ] Add check latest transfer operation samples for sts
- [ ] Add transfer from posix samples for sts
- [ ] Add transfer between posix
- [ ] Add download to posix
- [ ] Add Event driven GCS

### CODE REVIEW

File: storagetransfer/src/posix_download.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ * Creates a request to transfer from the local file system to the sink bucket
+ *
+ * @param string $projectId Your Google Cloud project ID.
+ * @param string $sinkAgentPoolName The agent pool associated with the POSIX data sink. Defaults to the default agent
+ * @param string $gcsSourceBucket Google Cloud Storage source bucket name.
+ * @param string $gcsSourcePath An optional path on the Google Cloud Storage bucket to download from.
+ * @param string $rootDirectory The root directory path on the destination filesystem.
+ */
+function posix_download($projectId, $sinkAgentPoolName, $gcsSourceBucket, $gcsSourcePath, $rootDirectory)
+{
```
Comment (bshaffer) [inline storagetransfer/src/posix_download.php:41]: add typehints 
```suggestion
function posix_download(
    string $projectId,
    string $sinkAgentPoolName,
    string $gcsSourceBucket,
    string $gcsSourcePath,
    string $rootDirectory
): void {
```

Comment (thiyaguk09) [reply storagetransfer/src/posix_download.php]: changes added

**Comment by danielduhh** [comment]
@bshaffer looks like bigtable is causing [static analysis](https://github.com/GoogleCloudPlatform/php-docs-samples/actions/runs/11740116258/job/33119526766?pr=2059#step:6:624) to fail. Any thoughts on mitigation?

File: storagetransfer/src/nearline_request.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    // $project = 'my-project-id';
+    // $description = 'My transfer job';
+    // $sourceGcsBucketName = 'my-source-bucket';
+    // $sinkGcsBucketName = 'my-sink-bucket';
+    // $startDate = new DateTime();
+
+    $dateTime = new DateTime($startDate);
+    $date = new Date([
+        'year' => $dateTime->format('Y'),
+        'month' => $dateTime->format('m') + 1,
```
Comment (bshaffer) [inline storagetransfer/src/nearline_request.php:62]: this is wrong, you do not need the +1

```suggestion
        'month' => $dateTime->format('m'),
```

**Review by bshaffer — APPROVED** [review]
I've made the necessary changes in [2248b46](https://github.com/GoogleCloudPlatform/php-docs-samples/pull/2059/commits/2248b4603961910853ded1a0a33c4806e8ab253b), so these tests are now passing.

In the future, please submit pull requests from _feature branches_, rather than from `main`. For instance, you could create a branch named `storage-transfer-samples` and submit the PR to `main` from there.

Thank you!

**Pull request merged**